From 6025951622cdc05e72201968c75dcdc5bcb1b7e4 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Sun, 3 Jan 2016 20:16:45 +0800
Subject: [PATCH 2/2] watchdog: octeon-wdt: Use PERCPU hander for percpu
 watchdog interrupts.

The octeon_wdt_poke_irq for each core must be set IRQF_PERCPU to
make sure that it can't be immigrated to other cores.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/watchdog/octeon-wdt-main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/watchdog/octeon-wdt-main.c b/drivers/watchdog/octeon-wdt-main.c
index 2a442a3..c32aff1 100644
--- a/drivers/watchdog/octeon-wdt-main.c
+++ b/drivers/watchdog/octeon-wdt-main.c
@@ -356,7 +356,7 @@ static void octeon_wdt_setup_interrupt(int cpu)
 		irq = OCTEON_IRQ_WDOG0 + core;
 
 	if (request_irq(irq, octeon_wdt_poke_irq,
-			IRQF_NO_THREAD, "octeon_wdt", octeon_wdt_poke_irq))
+			IRQF_NO_THREAD | IRQF_PERCPU, "octeon_wdt", octeon_wdt_poke_irq))
 		panic("octeon_wdt: Couldn't obtain irq %d", irq);
 
 	/* Must set the irq affinity here */
-- 
1.7.5.4

